utilities/ImpactUtilities.groovy

```diff
@@ -298,6 +308,110 @@ def scanOnlyStaticDependencies(List buildList, RepositoryClient repositoryClient
+				repositoryClient.getAllCollections().each{ collection ->
+					String cName = collection.getName()
+					if(matchesPattern(cName,collectionMatcherPatterns)){ // find matching collection names
+						if (cName != props.applicationCollectionName && cName != props.applicationOutputsCollectionName){
+							def Set<String> externalImpactList = collectionImpactsSetMap.get(cName) ?: new HashSet<String>()
+							def logicalFiles = repositoryClient.getAllLogicalFiles(cName, ldepFile);
+							logicalFiles.each{ logicalFile ->
+								def impactRecord = "${logicalFile.getLname()} \t ${logicalFile.getFile()} \t ${cName}"
+								// if (props.verbose) println("*** $impactRecord")
+								externalImpactList.add(impactRecord)
+							}
+							collectionImpactsSetMap.put(cName, externalImpactList)
+						}
+					}
+					else{
+						//if (props.verbose) println("$cName does not match pattern: $collectionMatcherPatterns")
+					}
+				}
+			}
+			else if(props.reportExternalImpactsAnalysisDepths == "deep"){
+				// Recursive analysis to support nested scenarios
+
+				// Configure impact resolver
+				ImpactResolver impactResolver = new ImpactResolver().file(changedFile).repositoryClient(repositoryClient)
+
+				String impactResolutionRules = props.getFileProperty('impactResolutionRules', changedFile)
+				impactResolver.setResolutionRules(buildUtils.parseResolutionRules(impactResolutionRules))
+
+				repositoryClient.getAllCollections().each{ collection ->
+					String cName = collection.getName()
+					if(matchesPattern(cName,collectionMatcherPatterns)){ // find matching collection names
+						if (cName != props.applicationCollectionName && cName != props.applicationOutputsCollectionName){
+							impactResolver.addCollection(cName) // add collection of foreign application
+						}
+					}
+					else{
+						//if (props.verbose) println("$cName does not match pattern: $collectionMatcherPatterns")
+					}
+				}
+				// resolve external impacted files
+				def externalImpactedFiles = impactResolver.resolve()
+
+				// report scanning results
+				if (externalImpactedFiles.size()!=0) if (props.verbose) println("*** Identified external impacted files for changed file $changedFile")
+				externalImpactedFiles.each{ externalImpact ->
+					def Set<String> externalImpactList = collectionImpactsSetMap.get(externalImpact.getCollection()) ?: new HashSet<String>()
+					def impactRecord = "${externalImpact.getLname()} \t ${externalImpact.getFile()} \t ${externalImpact.getCollection()}"
+					// if (props.verbose) println("*** $impactRecord")
+					externalImpactList.add(impactRecord)
+					collectionImpactsSetMap.put(externalImpact.getCollection(), externalImpactList) // <collection,list of impacted files>
+				}
+			}
+			else {
+				println("*! build property reportExternalImpactsAnalysisDepths has in invalid value : ${props.reportExternalImpactsAnaylsisDepths} , valid: simple | deep")
+			}
```

Comment:
I think that this control should be in build.groovy during parameters validation.

Reply:
Added an assertion